services.py

```
@@ -0,0 +1,301 @@
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+                bufsize=1
+            )
+            
+            # Collect all output for potential error display
+            all_output = []
+            line_count = 0
+            max_display = 20
+            
+            assert process.stdout is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)